source/addonStore/dataManager.py

```diff
@@ -162,6 +169,49 @@ def _cacheCompatibleAddons(self, addonData: str, cacheHash: Optional[str]):
 		with open(self._cacheCompatibleFile, "w", encoding="utf-8") as cacheFile:
 			json.dump(cacheData, cacheFile, ensure_ascii=False)
 
+	def _shouldCacheCompatibleAddonsBackup(self) -> bool:
+		if not os.path.exists(self._cacheCompatibleOldFile):
+			return True
+		resetNewAddons = config.conf["addonStore"]["resetNewAddons"]
+		if resetNewAddons == "startup":
+			return True
+		lastBackupTime = os.path.getmtime(self._cacheCompatibleOldFile)
+		lastBackupDate = datetime.fromtimestamp(lastBackupTime)
+		nowDate = datetime.now()
+		diffDate = nowDate - lastBackupDate
+		if resetNewAddons == "weekly" and diffDate.days >= 7:
+			return True
+		if resetNewAddons == "monthly" and diffDate.days >= 30:
+			return True
+		return False
+
+	def _getResetNewAddonsDate(self) -> str:
+		resetNewAddons = config.conf["addonStore"]["resetNewAddons"]
+		if resetNewAddons == "startup":
+			# Translators: Message presented in the new add-ons tab, informing that new add-ons will be reset at startup
+			return _("Will be reset at startup")
+		lastBackupTime = os.path.getmtime(self._cacheCompatibleOldFile)
+		lastBackupDate = datetime.fromtimestamp(lastBackupTime)
+		formattedLastBackupDate = lastBackupDate.strftime("%d/%m/%Y")
+		if resetNewAddons == "monthly":
+			timedeltaDays = 30
```

Comment:
More months have 31 days, maybe use that?

```suggestion
			timedeltaDays = 31
```